<?xml version="1.0"?>
<launch>
    <!-- Arguments for the launcher -->
    <arg name="ns"          default="" />
    <arg name="map_name"    default="map.yaml" />

    <!--
        Define the frame names to use for the robot
    -->
    <arg name="odom_frame"      default="$(arg ns)odom" />
    <arg name="base_frame"      default="$(arg ns)base_link" />
    <arg name="camera_frame"    default="$(arg ns)camera_link" />
    <arg name="imu_frame"       default="$(arg ns)imu_link" />
    <arg name="laser_frame"     default="$(arg ns)laser_link" />

    <!--
        My Nodes
        - Localization and Detect Aruco Nodes
    -->
        <!-- Detect Aruco Markers Node-->
        <!-- <node pkg="detect_aruco_markers" type="detect_aruco_markers_node" name="detect_aruco_markers" >
        </node> -->

        <!-- Localization Node -->
        <node pkg="localization_aruco" type="localization_aruco_node" name="localization_aruco" output="screen" >
            <param name="camera_frame" type="string" value="$(arg camera_frame)" />
            <param name="base_frame" type="string" value="$(arg base_frame)" />
            <param name="map_file_path" type="string" value="$(find localization_aruco)/maps/" />
            <param name="map_file_name" type="string" value="$(arg map_name)" />
        </node>


        <!-- Node attributes -->
        <!--
            <node
                pkg="package-name"
                type="name-of-executable"
                name="node-name"
                args="arg1 arg2 arg3"
                machine="machine-name-where-to-launch-node"
                respawn="true-or-false__restart_node_if_it_dies"
                respawn_delay="time-in-seconds__time_to_wait_before_restarting_node"
                required="true-or-false__if-true-then-if-node-dies-then-kill-all-nodes"
                ns="namespace__start-node-in-this-namespace"
                clear_params="true-or-false__delete-all-parameters-in-node's-private-namespace-before-launch"
                output="screen-or-log__where-to-print-node-output"
                cwd="ROS_HOME|node____set-current-working-directory__if 'node' then cwd-is-node's-executable-directory"
                launch-prefix="prefix arguments____command/arguments-to-prepend-to-node's-launch-arguments__ex:_gdb__valgrind__etc"
                if="true-or-false___used-to-evaluate-arguments/parameters-and-run-based-on-outcome-without-modifying-launch-file"
            >
                <env name="name" value="value" />
                <remap from="from" to="to" />
                <param name="name" value="value" ___Set a parameter in the node's ~/local namespace___ />
                <rosparam command="load|dump" file="filename" __Load-a-rosparam-file-into-this-node's ~/local namespace___ />
            </node>

        -->


</launch>